client-src/elements/chromedash-add-stage-dialog.js

```diff
@@ -0,0 +1,123 @@
+import {LitElement, css, html} from 'lit';
+import {SHARED_STYLES} from '../sass/shared-css.js';
+import {CREATEABLE_STAGES, FORMS_BY_STAGE_TYPE} from './form-definition.js';
+
+
+let AddStageDialogEl;
+
+
+export async function openAddStageDialog(featureId, featureType) {
+  if (!AddStageDialogEl) {
+    AddStageDialogEl = document.createElement('chromedash-add-stage-dialog');
+    AddStageDialogEl.featureId = featureId;
```

Comment:
This only sets the featureId when the chromedash-add-stage-dialog is first created.   If the user clicks away to close the dialog, then reopens it on a different feature, it would add a stage to the original feature.

Reply:
Good catch. I added a check that saves the `featureId` and hits this block of code if a new feature ID is passed in.